### PR TITLE
Feature: form request status validation

### PR DESCRIPTION
**Pull request by jonwaldstein**
Resolves [GIVE-1012]

## Description

This adds additional donation form status validation to the donate request.

## Affects

Donation form processing

## Visuals

Screenshot 2024-07-12 at 7 40 11 PM

## Testing Instructions

TBD

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-1012]: https://stellarwp.atlassian.net/browse/GIVE-1012?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/DonationForms/DataTransferObjects/DonateFormRouteData.php [hunk 1]
```diff
@@ -134,4 +136,20 @@ public function toArray(): array
...
+    /**
+     * @unreleased
+     */
+    private function isValidForm(DonationForm $form): bool
+    {
+        if ($form->status->isTrash()) {
+            return false;
+        }
+
+        if (!$form->status->isPublished() && !current_user_can('edit_give_forms')) {
```
Comment (pauloiankoski) [inline src/DonationForms/DataTransferObjects/DonateFormRouteData.php:149]: I just want to confirm that your intention here is to allow admin users to still be able to donate on unpublished forms.

**Review by pauloiankoski — APPROVED** [review]
@jonwaldstein I left a comment, but everything else looks great. I’m approving this PR.

File: src/DonationForms/DataTransferObjects/DonateFormRouteData.php (hunk 1 above)
Comment (jonwaldstein) [reply src/DonationForms/DataTransferObjects/DonateFormRouteData.php:149]: @pauloiankoski yes that's correct or at least the intention. Published forms should allow public requests.  If the form is in pending/draft/private mode than it should only be used by an admin who is likely using for testing.

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**